pkg/kube/inject/webhook.go

```diff
@@ -1153,6 +1153,22 @@ func (wh *Webhook) inject(ar *kube.AdmissionReview, path string) *kube.Admission
 		} else {
 			log.Warnf("unable to fetch namespace, failed to get client for %q", clusterID)
 		}
+
+		// OpenShift automatically assigns a SecurityContext.RunAsUser to all containers in the Pod, even if the Pod's
+		// YAML does not explicitly set this value. Istio treats the values specified in the istio-proxy container as
+		// overrides and preserves them in the final Pod yaml as expected. However, the RunAsUser value which is
+		// automatically set by OpenShift would be the same for all containers within the Pod, which is a problem.
```

Comment:
```suggestion
		// automatically set by OpenShift would be the same for all containers within the Pod, which is a problem.
```

Is there no way in openshift to denylist specific containers from blanket RunAsUser overrides?

Reply:
It's not so much a UX bug rather than a UX feature: if you don't set a UID in the correct range, your pods will fail to start because of the PodSecurityPolicy that is used by default ('restricted') - it forces you to use UIDs from a range that OpenShift writes into your namespace's annotations. But OpenShift doesn't know that the pods should have different UIDs and expects users to set them manually if they do require that. I guess a fix on the OpenShift side could be to assign unique IDs per container by default

Reply:
> to the first legal value in the range, for all containers. But OpenShift doesn't know that the pods should have different UIDs and expects users to set them manually if they do require that. I guess a fix on the OpenShift side could be to assign unique IDs per container by default

Right - my confusion stems from the fact that we _do_ set the UID for one container manually, but that is being wiped out because it's all-or-nothing in terms of Openshift assignment - assigning UIDs to every _container_ that didn't already have one defined manually would be a good improvement.

But I'm not against doing this here if we have to.

Reply:
> I would not expect OpenShift to override the explicit UID I have defined for one container, just because the other containers in the pod have not defined one, but it looks like that is the behavior currently.

I'm afraid that's not the case. In OCP deployments, we can explicitly set runAsUser (as long as its in the range assigned to the namespace) for **specific containers** in the deployment, and OCP will respect this setting while assigning a runAsUser to other containers where it is not specified. In this PR, our main goal is to ensure that the runAsUser for the sidecar proxy does not conflict with the one automatically assigned by OCP.